### PR TITLE
Added the ability to get locks on multiple resources

### PR DESCRIPTION
**Pull request by GrahamCampbell**

### CODE REVIEW

**Comment by joecohens** [comment]
:+1:

File: src/Lock.php [hunk 1]
```diff
@@ -130,10 +138,24 @@ public function acquire()
         while (true) {
             $this->token = str_random(32);
 
-            if ($this->redis->set($this->name, $this->token, 'NX', 'PX', $this->timeout)) {
+            $acquired = [];
+
+            foreach ($this->names as $name) {
+                if ($this->redis->set($this->names, $this->token, 'NX', 'PX', $this->timeout)) {
```
Comment (GrahamCampbell) [inline src/Lock.php]: typo here

Comment (GrahamCampbell) [inline src/Lock.php]: should be `$name` not `$this->names`

**Comment by GrahamCampbell** [comment]
Rebased.

**Pull request closed without merging**